### PR TITLE
Add hard coded filter by work item type feature (#882 and #658)

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Display types filter option on the side panel. Clicking on the type filter will update the work item list.
For now, we have hard coded User Story and Bug work item type.

This is a functional implementation only.

### CODE REVIEW

File: src/app/iteration/iteration.component.scss [hunk 1]
```diff
@@ -14,8 +14,10 @@
   -webkit-flex-direction: column;
   -ms-flex-direction: column;
   flex-direction: column;
+  /* Commenting this for hard coding the types value
   height: calc(100% - 190px) !important; // height is given to scroll the content
-  overflow: scroll !important; // important is given to override collapse module styles
+  */
+  overflow: auto; // important is given to override collapse module styles
```
Comment (debloper) [inline src/app/iteration/iteration.component.scss]: I guess it's okay to remove the inline comment now.

File: src/app/side-panel/side-panel.component.ts [hunk 2]
```diff
@@ -11,10 +12,20 @@ export class SidepanelComponent implements OnInit {
...
   constructor(
-    private router: Router) {
+    private router: Router,
+    private broadcaster: Broadcaster) {
   }
 
   ngOnInit() {
   }
 
+   getWorkItemsByIteration(iteration: any) {
```
Comment (debloper) [inline src/app/side-panel/side-panel.component.ts]: nit: indentation

File: src/app/side-panel/side-panel.component.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { Component, OnInit } from '@angular/core';
 import { Router }            from '@angular/router';
+import { Broadcaster } from './../shared/broadcaster.service';
```
Comment (debloper) [inline src/app/side-panel/side-panel.component.ts]: nit: declaration alignment

File: src/app/type/type.component.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<!--Type Start-->
+<div class="type-panel-header pointer display-flex">
+  <h3 class="flex-1">Types</h3>
```
Comment (debloper) [inline src/app/type/type.component.html]: Don't think `class="flex-1"` is necessary, cause all it does is sets `flex: 1` which is always the default anyways.

File: src/app/type/type.component.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+</div>
+<div class="types-container display-flex">
+  <div class="type-wrapper">
+    <div class="types">
+      <ul class="not-current-type type-expanded-div">
+        <li class="type-subheader display-flex">
+          <div class="flex-1 truncate">
```
Comment (debloper) [inline src/app/type/type.component.html]: Same, `class="flex-1"` not needed.

File: src/app/type/type.component.html [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+            <strong class="type-name "
+              tooltip="User Story"
+              tooltipPlacement="top"
+              (click)='getWorkItemsByType("userstory")'>
+              User Story
+            </strong>
+          </div>
+        </li>
+        <li class="type-subheader display-flex">
+          <div class="flex-1 truncate">
```
Comment (debloper) [inline src/app/type/type.component.html]: `class="flex-1"` not needed.

File: src/app/type/type.component.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import { Observable } from 'rxjs/Observable';
+import { Subscription } from 'rxjs/Subscription';
+
+import { Params, ActivatedRoute } from '@angular/router';
+import { SpaceService, Space } from './../shared/mock-spaces.service';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+
+import { AuthenticationService } from './../auth/authentication.service';
```
Comment (debloper) [inline src/app/type/type.component.ts]: Do we need auth service here?
Let's add it when we need it.

**Comment by Mgranfie** [comment]
@nimishamukherjee I am not following what this functional implementation is referring to. Can you clarify for me? What feature and behaviors/tasks does this relate to?

**Pull request merged**